GitCommands/Git/GitModule.cs

```diff
@@ -2861,41 +2861,42 @@ public RemoteActionResult<IReadOnlyList<IGitRef>> GetRemoteServerRefs(string rem
             return result;
         }
 
-        public IReadOnlyList<IGitRef> GetRefs(bool tags = true, bool branches = true)
+        public IReadOnlyList<IGitRef> GetRefs(bool tags = true, bool branches = true, GitRefsSortBy sortBy = GitRefsSortBy.Default, GitRefsSortOrder sortOrder = GitRefsSortOrder.Ascending)
         {
-            return GetRefs(tags, branches, false);
+            return GetRefs(tags, branches, noLocks: false, sortBy, sortOrder);
         }
 
-        public IReadOnlyList<IGitRef> GetRefs(bool tags, bool branches, bool noLocks)
+        public IReadOnlyList<IGitRef> GetRefs(bool tags, bool branches, bool noLocks, GitRefsSortBy sortBy, GitRefsSortOrder sortOrder)
         {
-            var refList = _gitExecutable.GetOutput(GetRefsCmd(tags: tags, branches: branches, noLocks: noLocks));
+            var refList = _gitExecutable.GetOutput(GetRefsCmd(tags, branches, noLocks, sortBy, sortOrder));
 
             return ParseRefs(refList);
         }
 
-        internal GitArgumentBuilder GetRefsCmd(bool tags, bool branches, bool noLocks)
+        internal GitArgumentBuilder GetRefsCmd(bool tags, bool branches, bool noLocks, GitRefsSortBy sortBy, GitRefsSortOrder sortOrder)
         {
             GitArgumentBuilder cmd;
 
             if (tags)
             {
-                cmd = new GitArgumentBuilder("show-ref", gitOptions:
-                    noLocks && GitVersion.Current.SupportNoOptionalLocks
-                        ? (ArgumentString)"--no-optional-locks"
-                        : default)
+                cmd = new GitArgumentBuilder("show-ref",
+                        gitOptions: noLocks && GitVersion.Current.SupportNoOptionalLocks
+                            ? (ArgumentString)"--no-optional-locks"
+                            : default)
                 {
                     { branches, "--dereference", "--tags" },
+                    { sortBy != GitRefsSortBy.Default, GetSortCriteria(sortBy, sortOrder), string.Empty }
                 };
             }
             else if (branches)
             {
                 // branches only
-                cmd = new GitArgumentBuilder("for-each-ref", gitOptions:
-                    noLocks && GitVersion.Current.SupportNoOptionalLocks
-                        ? (ArgumentString)"--no-optional-locks"
-                        : default)
+                cmd = new GitArgumentBuilder("for-each-ref",
+                        gitOptions: noLocks && GitVersion.Current.SupportNoOptionalLocks
+                            ? (ArgumentString)"--no-optional-locks"
+                            : default)
                 {
-                    "--sort=-committerdate",
+                    { sortBy != GitRefsSortBy.Default, GetSortCriteria(sortBy, sortOrder), string.Empty },
```

Comment:
--sort was introduced in Git 2.14.6 (not 2.19, not opposing the raise)
For all other Git options, GE checks arguments to avoid crashes. Should that be done here?

Reply:
Yes, but `git branch.sort` option was only introduced in 2.19
https://github.blog/2018-09-10-highlights-from-git-2-19/#user-content-sorting-branches

> For all other Git options, GE checks arguments to avoid crashes. Should that be done here?

Please elaborate, I don't think I understand.

Reply:
> Yes, but git branch.sort option was only introduced in 2.19

Not what is used here, so min version is 2.14.6

> Please elaborate, I don't think I understand.

GE has checks for Git options in versions, also for unsupported versions to avoid crashes.

Reply:
I'm split. We're moving the bar up, so adding checks for something we no longer support feels redundant.
I'll add it if you feel it is truly necessary.